src/filters.ts

```diff
@@ -217,12 +218,27 @@ angular.module('3ema.filters', [])
 })
 
 /**
- * Return whether contact is not me.
+ * Filter receivers, exclude own contact.
  */
 .filter('isNotMe', ['WebClientService', function(webClientService: WebClientService) {
-    return function(obj: threema.Receiver) {
+    return function(receivers: threema.Receiver[]) {
         const valid = (contact: threema.Receiver) => contact.id !== webClientService.receivers.me.id;
-        return filter(obj, valid);
+        return filter(receivers, valid);
+    };
+}])
+
+/**
+ * Filter receivers, exclude hidden contact.
+ */
+.filter('isNotHidden', [function() {
+    return function(receivers: threema.Receiver) {
```

Comment:
(Reply to https://github.com/threema-ch/threema-web/pull/661/files#r229691268)

This is a function that accepts some kind of iterable. `filter` accepts a list or an object and filters it with the given callback. Thus, unless there is some magic going on, each filter will go through the whole list.

Reply:
Ah *that's* what you mean :slightly_smiling_face: You are right, sorry. I'll fix!

Reply:
Fixed in the fixup commit. I decided to ditch the filters altogether and to reuse the iterator we already have.

Reply:
Yeah, I think this filter is ancient and IIRC has been used in other places previously.